[Bug] Breaking change: BookLore crashes on startup without /bookdrop volume mount after f4d1ba7

### What Went Wrong?

This is a breaking change introduced by commit f4d1ba7 ("feat(bookdrop): add ability to poll the bookdrop folder"). After updating to a build that includes this commit, BookLore will crash on startup if the `/bookdrop` volume is not mounted in docker-compose.yml. Any user upgrading to this release without the `/bookdrop` volume mount will have a non-functional container that restart-loops indefinitely. This is not documented as a breaking change in the release notes, so users upgrading will hit this unexpectedly.

The new `BookdropPeriodicScanTask` component is always instantiated by Spring at startup, which pulls in `BookdropMonitoringService`, whose `@PostConstruct start()` method calls `Files.createDirectories("/bookdrop")`. Without the volume mount, the container process can't write to the filesystem root and the entire application context fails to load.

Previously, not mounting `/bookdrop` didn't prevent the app from starting.

### How Can We Reproduce This?

1. Upgrade BookLore to a version that includes commit f4d1ba7
2. Do not have a `/bookdrop` volume mount in docker-compose.yml
3. Container fails to start and enters a crash loop

### What Should Have Happened?

The app should start gracefully even without `/bookdrop` mounted. The `BookdropMonitoringService` could log a warning and skip initialization, or `BookdropPeriodicScanTask` could be conditionally registered only when the bookdrop path is accessible. At minimum, this should be documented as a breaking change so users know to add the volume mount before upgrading.

### Screenshots or Error Messages (Optional)

```
Caused by: java.nio.file.AccessDeniedException: /bookdrop
    at org.booklore.service.bookdrop.BookdropMonitoringService.start(BookdropMonitoringService.java:49)
```

Full dependency chain: `taskService` → `bookdropPeriodicScanTask` → `bookdropMonitoringService` → @PostConstruct init fails

### Any Ideas on How to Fix This? (Optional)

The migration in V122 adds the cron job as disabled by default (`enabled = FALSE`), but since `BookdropPeriodicScanTask` is a `@Component`, Spring still instantiates the bean regardless of whether the cron is enabled, triggering the full init chain and the crash.

Workaround: add `- ./bookdrop:/bookdrop` to volumes in docker-compose.yml before upgrading.

- Wrap the `@PostConstruct start()` in `BookdropMonitoringService` with a try/catch around `Files.createDirectories()` so it logs a warning and skips initialization instead of crashing the app
- Use `@ConditionalOnExpression` or similar to only register `BookdropPeriodicScanTask` when the bookdrop path exists and is writable
- Create `/bookdrop` in the Dockerfile so the directory always exists inside the container, even without a volume mount

### Your Setup

- Booklore version: latest (includes f4d1ba7)
- Spring Boot 4.0.3 / Spring 7.0.5
- How did you install Booklore? Docker on Unraid

### Before Submitting

- [x] I've searched existing issues and confirmed this bug hasn't been reported yet

--- a/Dockerfile
+++ b/Dockerfile
@@ -56,7 +56,8 @@ LABEL org.opencontainers.image.title="BookLore" \
 
 ENV JAVA_TOOL_OPTIONS="-XX:+UseG1GC -XX:MaxGCPauseMillis=200 -XX:+UseStringDeduplication -XX:+UseContainerSupport -XX:+UseCompactObjectHeaders -XX:MaxRAMPercentage=75.0"
 
-RUN apk update && apk add --no-cache su-exec
+RUN apk update && apk add --no-cache su-exec && \
+    mkdir -p /bookdrop
 
 COPY entrypoint.sh /usr/local/bin/entrypoint.sh
 RUN chmod +x /usr/local/bin/entrypoint.sh

--- a/booklore-api/src/main/java/org/booklore/service/bookdrop/BookdropMonitoringService.java
+++ b/booklore-api/src/main/java/org/booklore/service/bookdrop/BookdropMonitoringService.java
@@ -30,6 +30,7 @@ public class BookdropMonitoringService {
     private volatile boolean running;
     private WatchKey watchKey;
     private volatile boolean paused;
+    private volatile boolean disabled;
 
     public BookdropMonitoringService(
             AppProperties appProperties,
@@ -42,29 +43,36 @@ public class BookdropMonitoringService {
     }
 
     @PostConstruct
-    public void start() throws IOException {
+    public void start() {
         bookdrop = Path.of(appProperties.getBookdropFolder());
         if (Files.notExists(bookdrop)) {
             try {
                 Files.createDirectories(bookdrop);
                 log.info("Created missing bookdrop folder: {}", bookdrop);
             } catch (IOException e) {
-                log.error("Failed to create bookdrop folder: {}", bookdrop, e);
-                throw e;
+                log.warn("Bookdrop folder is not available at '{}'. Bookdrop monitoring is disabled. " +
+                        "Mount a volume at this path to enable it.", bookdrop);
+                this.disabled = true;
+                return;
             }
         }
 
-        log.info("Starting bookdrop folder monitor: {}", bookdrop);
-        this.watchService = FileSystems.getDefault().newWatchService();
-        this.watchKey = bookdrop.register(watchService,
-                StandardWatchEventKinds.ENTRY_CREATE,
-                StandardWatchEventKinds.ENTRY_DELETE);
-        this.running = true;
-        this.paused = false;
-        this.watchThread = new Thread(this::processEvents, "BookdropFolderWatcher");
-        this.watchThread.setDaemon(true);
-        this.watchThread.start();
-        scanExistingBookdropFiles();
+        try {
+            log.info("Starting bookdrop folder monitor: {}", bookdrop);
+            this.watchService = FileSystems.getDefault().newWatchService();
+            this.watchKey = bookdrop.register(watchService,
+                    StandardWatchEventKinds.ENTRY_CREATE,
+                    StandardWatchEventKinds.ENTRY_DELETE);
+            this.running = true;
+            this.paused = false;
+            this.watchThread = new Thread(this::processEvents, "BookdropFolderWatcher");
+            this.watchThread.setDaemon(true);
+            this.watchThread.start();
+            scanExistingBookdropFiles();
+        } catch (IOException e) {
+            log.warn("Failed to start bookdrop folder monitor. Bookdrop monitoring is disabled.", e);
+            this.disabled = true;
+        }
     }
 
     @PreDestroy
@@ -84,6 +92,7 @@ public class BookdropMonitoringService {
     }
 
     public synchronized void pauseMonitoring() {
+        if (disabled) return;
         if (!paused) {
             if (watchKey != null) {
                 watchKey.cancel();
@@ -97,6 +106,7 @@ public class BookdropMonitoringService {
     }
 
     public synchronized void resumeMonitoring() {
+        if (disabled) return;
         if (paused) {
             try {
                 watchKey = bookdrop.register(watchService,
@@ -190,6 +200,10 @@ public class BookdropMonitoringService {
     }
 
     public void rescanBookdropFolder() {
+        if (disabled) {
+            log.warn("Bookdrop monitoring is disabled. Skipping rescan.");
+            return;
+        }
         log.info("Rescan of Bookdrop folder triggered.");
         scanExistingBookdropFiles();
     }

